### PR TITLE
Fix MoveAdjacentToElement Error

### LINKED ISSUE #188: Unexpected Error Occurred
Worked fine this morning. Launched in the afternoon, I get this after the splash screen.
This is Win 10 Pro

Saturday, December 19, 2015 5:43:38 PM

Version: 0.5.1.2

OS Version: Microsoft Windows NT 6.2.9200.0
Runtime version: 4.0.30319.42000
Shutdown started: False
Program: "C:\Users\Bogey\AppData\Local\OpenLiveWriter\app-0.5.1.2\OpenLiveWriter.exe" 

Memory Load: 64%
Total Physical: 1023 MB
Available Physical: 360 MB
Total Page File: 2751 MB
Available Page File: 1969 MB
Total Virtual: 2047 MB
Available Virtual: 1697 MB
Available Extented Virtual: 0 MB

Unexpected Error Occurred
An unexpected error has occurred within the application.

System.ArgumentException: Value does not fall within the expected range.
   at OpenLiveWriter.Mshtml.IMarkupPointerRaw.MoveAdjacentToElement(IHTMLElement pElement, _ELEMENT_ADJACENCY eAdj)
   at OpenLiveWriter.Mshtml.MshtmlMarkupServices.CreateMarkupRange(IHTMLElement element, Boolean outside)
   at OpenLiveWriter.HtmlEditor.HtmlEditorControl._mshtmlEditor_DocumentComplete(Object sender, EventArgs e)
   at OpenLiveWriter.Mshtml.MshtmlEditor.OnDocumentComplete()
   at OpenLiveWriter.Mshtml.MshtmlEditor.ReadyStateChangedHandler(Object sender, EventArgs ea)

### PR DESCRIPTION
**Pull request by ScottIsAFool**
Fixes #188

### CODE REVIEW

**Comment by willduff** [comment]
Looks good, but it'd be worth looking at who is calling it with `null` and why.

**Comment by willduff** [comment]
Actually, thinking about this more, I'd rather get this null pointer exception when its called incorrectly, we should really fix the caller instead.

Looks like `PostBodyElement` is null in `HtmlEditorControl.cs` based on the callstack in #188. Can you add an additional check here?
https://github.com/OpenLiveWriter/OpenLiveWriter/blob/master/src/managed/OpenLiveWriter.HtmlEditor/HtmlEditorControl.cs#L1189
`if (DocumentIsReady && PostBodyElement != null)`

**Comment by ScottIsAFool** [comment]
Added the additional check

File: src/managed/OpenLiveWriter.HtmlEditor/HtmlEditorControl.cs [hunk 1]
```diff
@@ -1186,7 +1186,7 @@ private void _mshtmlEditor_DocumentComplete(object sender, EventArgs e)
             OnDocumentComplete(e);
 
             // Remove null-attributed font tags, e.g. <p><font>blah</font></p> --> <p>blah</p>
-            if (DocumentIsReady)
+            if (DocumentIsReady & PostBodyElement != null)
```
Comment (willduff) [inline src/managed/OpenLiveWriter.HtmlEditor/HtmlEditorControl.cs]: &&

Comment (ScottIsAFool) [inline src/managed/OpenLiveWriter.HtmlEditor/HtmlEditorControl.cs]: Waaaaaaay too early in the morning :) Have corrected.

File: src/managed/OpenLiveWriter.Mshtml/MarkupPointer.cs [hunk 2]
```diff
@@ -336,7 +336,10 @@ public IHTMLElement SeekElementRight(IHTMLElementFilter filter, MarkupPointer bo
         /// <param name="eAdj"></param>
         public void MoveAdjacentToElement(IHTMLElement element, _ELEMENT_ADJACENCY eAdj)
         {
-            PointerRaw.MoveAdjacentToElement(element, eAdj);
+            if (element != null)
```
Comment (willduff) [inline src/managed/OpenLiveWriter.Mshtml/MarkupPointer.cs]: Sorry, but can you just revert the change here? I actually want it to crash :smile:. We are getting good crash data from this and there are a couple other instances we should go and fix too.

**Pull request merged**